Assets/Scripts/SS3D/Systems/Entities/Humanoid/Ragdoll.cs

```diff
@@ -0,0 +1,334 @@
+            _standUpBones = new BoneTransform[_ragdollParts.Length];
+            _ragdollBones = new BoneTransform[_ragdollParts.Length];
+
+            for (int boneIndex = 0; boneIndex < _ragdollParts.Length; boneIndex++)
+            {
+                _standUpBones[boneIndex] = new();
+                _ragdollBones[boneIndex] = new();
+            }
+            // All rigid bodies are kinematic at start, only the owner should be able to change that afterwards.
+			ToggleKinematic(true);
+		}
+        
+		private void Update()
+		{
+            if (IsServer && _isKnockdownTimed && IsKnockedDown)
+            {
+                _knockdownTimer -= Time.deltaTime;
+                if (_knockdownTimer <= 0)
+                {
+                    Recover();
+                }
+            }
+            switch (_currentState)
+            {
+                case RagdollState.Walking:
+                    WalkingBehavior();
+                    break;
+                case RagdollState.Ragdoll:
+                    RagdollBehavior();
+                    break;
+                case RagdollState.BonesReset:
+                    BonesResetBehavior();
+                    break;
+                case RagdollState.StandingUp:
+                    StandingUpBehavior();
+                    break;
+            }
+        }
+
+        private void WalkingBehavior() { }
+        /// <summary>
+        /// Cast knockdown, that isn't going to expire until Recover()
+        /// </summary>
+        [ServerRpc(RequireOwnership = false)]
+        public void KnockdownTimeless()
+        {
+            _isKnockdownTimed = false;
+            IsKnockedDown = true;
+        }
+        /// <summary>
+        /// Knockdown the character for some time.
+        /// </summary>
+        /// <param name="seconds"></param>
+        [ServerRpc(RequireOwnership = false)]
+        public void Knockdown(float seconds)
+        {
+            _isKnockdownTimed = true;
+            _knockdownTimer += seconds;
+            IsKnockedDown = true;
+        }
+        private void Knockdown()
+        {
+            _currentState = RagdollState.Ragdoll;
+            ToggleController(false);
+            ToggleAnimator(false);
+            
+            if (!IsOwner) return;
+            Vector3 movement = _humanoidLivingController.TargetMovement * 3;
+            ToggleKinematic(false);
+            foreach (Transform part in _ragdollParts)
+            {
+                part.GetComponent<Rigidbody>().AddForce(movement, ForceMode.VelocityChange);
+            }
+        }
+
+        private void RagdollBehavior()
+        {
+            // Only the owner handles ragdoll's physics
+            if (!IsOwner) return;
+            AlignToHips();
+        }
+        /// <summary>
+        /// Adjust player's position and rotation
+        /// </summary>
+        private void AlignToHips()
```

Comment:
Add more doc, this method is pretty complicated, I want to understand what the adjustment is for just by reading the doc